config/s3-no-setup.json

```diff
@@ -0,0 +1,85 @@
+{
+  "@context": "https://linkedsoftwaredependencies.org/bundles/npm/@solid/community-server/^5.0.0/components/context.jsonld",
+  "import": [
+    "css:config/app/main/default.json",
+    "css:config/app/init/initialize-root.json",
+    "css:config/app/setup/disabled.json",
+    "css:config/app/variables/default.json",
+    "css:config/http/handler/default.json",
+    "css:config/http/middleware/websockets.json",
+    "css:config/http/server-factory/websockets.json",
+    "css:config/http/static/default.json",
+    "css:config/identity/access/public.json",
+    "css:config/identity/email/default.json",
+    "css:config/identity/handler/default.json",
+    "css:config/identity/ownership/token.json",
+    "css:config/identity/pod/static.json",
+    "css:config/identity/registration/enabled.json",
+    "css:config/ldp/authentication/dpop-bearer.json",
+    "css:config/ldp/authorization/webacl.json",
+    "css:config/ldp/handler/default.json",
+    "css:config/ldp/metadata-parser/default.json",
+    "css:config/ldp/metadata-writer/default.json",
+    "css:config/ldp/modes/default.json",
+    "css:config/storage/backend/data-accessors/file.json",
+    "css:config/storage/backend/data-accessors/s3.json",
+    "css:config/storage/key-value/memory.json",
+    "css:config/storage/middleware/default.json",
+    "css:config/util/auxiliary/acl.json",
+    "css:config/util/identifiers/suffix.json",
+    "css:config/util/index/default.json",
+    "css:config/util/logging/winston.json",
+    "css:config/util/representation-conversion/default.json",
+    "css:config/util/resource-locker/debug-void.json",
+    "css:config/util/variables/default.json"
+  ],
+  "@graph": [
+    {
+      "comment": "A single-pod server that stores its resources on disk."
+    },
+    {
+      "comment": "A more complex example, routing data to 3 different stores.",
+      "@id": "urn:solid-server:default:ResourceStore_Backend",
+      "@type": "RoutingResourceStore",
+      "rule": { "@id": "urn:solid-server:default:RouterRule" }
+    },
+
+    {
+      "comment": [
+        "Configure routing to send all requests containing `/file/` to the file store, containing `/memory/` to the memory store, and `/sparql/` to the SPARQL endpoint store.",
```

Comment:
Thanks for spotting!

The reason is that ACL files are small and often needed, which is a scenario where S3 performs less well. More specifically, one would probably store those better in a separate KV- or SPARQL-store.

Same goes for metadata actually. I've now added these to S3, because it supports that natively, but the way the `DataAccessorBasedStore` handles requests makes it rather inefficient: it calls the `DataAccessor` separately for the data and the metadata, whilst both can be retreived together from S3.

Reply:
At some point we would like to split the `DataAccessorBasedStore` into multiple pieces so specific implementations can reuse the parts they need without depending entirely on the entire class but that's probably not going to happen soon. #998

Reply:
You could also introduce caching using a WeakMap, it's something we do in some places in the CSS already to prevent multiple calls to the backend. The `ResourceIdentifier` object will only exist for this action, so you could already request data+metadata on the first function call and then store it in the map to return on the subsequent `getData` call.